showtime-xyz/showtime-frontend
# Feature/show2 1116 FlashList applying to the Tab

# Why

- we want to get rid of `recyclerlistview`'s API.
- the FlashList has a great performance.
- we want to use `InfiniteScrollList` everywhere.

but the key tab list is not used yet,  so this PR is all for the Tabs list.


# How

- wrap an `Animated.ScrollView` to use in FlashList, 
- encapsulation our `InfiniteScrollList` components to replace old `TabRecyclerList`. 
- fixed some types and bugs.

# Test Plan

- check if the profile/trending screen looks good.
- check their performance.
- check if the list has some bugs.
